wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -248,6 +271,27 @@ def cancel(self) -> None:
         self.suspend()
         self.batch_api.delete_namespaced_job(name=self.name, namespace=self.namespace)
 
+    def cancel_with_warning(self) -> None:
+        try:
+            self.batch_api.delete_namespaced_job(
+                name=self.name,
+                namespace=self.namespace,
+                grace_period_seconds=0,
+                propagation_policy="Foreground",
+            )
+            wandb.termwarn(f"Job was cancelled: {self.name}")
+        except Exception:
+            pass
+
+    def is_cancelled(self) -> bool:
+        try:
+            self.core_api.read_namespaced_pod(
+                name=self.pod_names[0], namespace=self.namespace
+            )
+            return False
+        except Exception:
```

Comment:
The exception for missing pod is probably pretty specific. Also, can we also check the job is gone?